azkaban-common/src/main/java/azkaban/container/models/PodTemplateMergeUtils.java

```diff
@@ -42,95 +45,180 @@ private PodTemplateMergeUtils() {
+    }
+
+    // Override name from the podSpecContainer to templateContainer.
+    if (null != podSpecContainer.getName()) {
+      templateContainer.setName(podSpecContainer.getName());
+    }
+
+    // Override ImagePullPolicy from the corresponding podSpecContainer.
+    if (null != podSpecContainer.getImagePullPolicy()) {
+      templateContainer.setImagePullPolicy(podSpecContainer.getImagePullPolicy());
+    }
+
+    // Override Image from the corresponding podSpecContainer.
+    if (null != podSpecContainer.getImage()) {
+      templateContainer.setImage(podSpecContainer.getImage());
+    }
+
+    // Override Resources from the corresponding podSpecContainer.
+    if (null != podSpecContainer.getResources()) {
+      templateContainer.setResources(podSpecContainer.getResources());
     }
+
+    // Merge Volume Mounts with podSpecContainerVolumeMounts overriding
+    // templateContainerVolumeMounts.
+    final List<V1VolumeMount> templateContainerVolumeMounts =
+        templateContainer.getVolumeMounts();
+    final List<V1VolumeMount> podSpecContainerVolumeMounts =
+        podSpecContainer.getVolumeMounts();
+    final List<V1VolumeMount> mergedContainerVolumeMounts =
+        getMergedContainerVolumeMounts(
+            templateContainerVolumeMounts, podSpecContainerVolumeMounts);
+    templateContainer.setVolumeMounts(mergedContainerVolumeMounts);
   }
 
   /**
-   * Add those initContainers which are not already available in the podSpec
+   * This method merges VolumeMounts such that: 1) All VolumeMounts from the
+   * podSpecContainerVolumeMounts are added. 2) Only those VolumeMounts of
+   * templateContainerVolumeMounts are added which are not already part of
+   * podSpecContainerVolumeMounts.
    *
-   * @param podSpec     Already created podSpec using the {@link AzKubernetesV1SpecBuilder}
-   * @param podTemplate Instance of the class {@link AzKubernetesV1PodTemplate} to extract items
+   * @param templateContainerVolumeMounts List of VolumeMounts from templateContainer.
+   * @param podSpecContainerVolumeMounts  List of VolumeMounts from podSpecContainer.
+   * @return List of VolumeMounts after merging templateInitContainerVolumeMounts and
+   * podSpecInitContainerVolumeMounts.
    */
-  private static void mergeInitContainers(V1PodSpec podSpec,
-      AzKubernetesV1PodTemplate podTemplate) {
-    List<V1Container> podSpecInitContainers = podSpec.getInitContainers();
-    if (null != podSpecInitContainers) {
-      List<V1Container> templateInitContainers = podTemplate.getInitContainers(
-          tempInitContainer -> podSpecInitContainers.stream().map(V1Container::getName)
-              .noneMatch(name -> name.equals(tempInitContainer.getName())));
-      for (V1Container containerItem : templateInitContainers) {
-        podSpec.addInitContainersItem(containerItem);
-      }
+  private static List<V1VolumeMount> getMergedContainerVolumeMounts(
+      List<V1VolumeMount> templateContainerVolumeMounts,
```

Comment:
Please use final for input params not meant to be out params. Applies to all such methods.

Reply:
done